connector/failoverconnector/README.md

```diff
@@ -0,0 +1,79 @@
+| [Code Owners](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#becoming-a-code-owner)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         |
+
+[in development]:https://github.com/open-telemetry/opentelemetry-collector#in-development
+[contrib]: https://github.com/open-telemetry/opentelemetry-collector-releases/tree/main/distributions/otelcol-contrib
+
+## Supported Pipeline Types
+
+| [Exporter Pipeline Type] | [Receiver Pipeline Type] | [Stability Level] |
+| ------------------------ | ------------------------ |-------------------|
+| traces | traces | [In development]  |
+| metrics | metrics | [In development]           |
+| logs | logs | [In development]           |
+
+[Exporter Pipeline Type]: https://github.com/open-telemetry/opentelemetry-collector/blob/main/connector/README.md#exporter-pipeline-type
+[Receiver Pipeline Type]: https://github.com/open-telemetry/opentelemetry-collector/blob/main/connector/README.md#receiver-pipeline-type
+[Stability Level]: https://github.com/open-telemetry/opentelemetry-collector#stability-levels
+<!-- end autogenerated section -->
+
+Allows for health based routing between trace, metric, and log pipelines depending on the health of target downstream exporters.
+
+## Configuration
+
+If you are not already familiar with connectors, you may find it helpful to first visit the [Connectors README].
+
+The following settings are available:
+
+- `priority (required)`: list of pipeline level priorities in a 1 - n configuration, multiple pipelines can sit at a single priority level.
+- `retry_interval (required)`: the frequency at which the pipeline levels will attempt to reestablish connection with all higher priority levels
+- `retry_gap (required)`: the amount of time between trying two separate priority levels in a single retry_interval timeframe
+- `max_retry (required)`: the maximum retries per level
```

Comment:
Typically we try to provide reasonable defaults wherever possible, in order to make it as simple as possible to get started with the component. These fields all seem like they could have reasonable defaults.

Reply:
The idea is that it should retry a maximum of one unhealthy level before returning to the stable level to avoid back pressure through the collector. Especially with retries enabled on the exporters it could take several seconds for each level that is retried. If we’re on a low enough level and each higher priority level needs to be retried prior to returning to stability, that timeframe could become significant very quickly.